ambari-server/src/main/java/org/apache/ambari/server/state/Config.java

```diff
@@ -102,4 +106,14 @@
    * @return the cluster where this config belongs to
    */
   Cluster getCluster();
+
+  /**
+   * Encrypt and persist the configuration.
+   */
+  void encryptSensitiveDataAndSave(Encryptor<Config> configPropertiesEncryptor);
+
+  /**
+   * Decrypt and persist the configuration.
+   */
+  void decryptSensitiveDataAndSave(Encryptor<Config> configPropertiesEncryptor);
```

Comment:
I'm not sure if we need this new API.
Within SensitiveDataEncryption you have everything you need to encrypt/decrypt. For instance:
              if (encrypt) {
                configEncryptor.encryptSensitiveData(config)
              } else {
                configEncryptor.decryptSensitiveData(config)
              }
              config.save()

What do you think?